### PR TITLE
SALTO-3549 - Zendesk Adapter - Default group change validator

### PR DESCRIPTION
**Pull request by idoser96**
Whenever a user will try to make a change of the default group, he will receive an error and an explanation of what to do
Possible cases:
1. Change of the default group
2. Addition of a group with it being the default
3. Removal of a group that is currently set as the default

---

None

---
_Release Notes_: 
Zendesk Adapter:
- Users will not be able to deploy any changes that includes the change of the default group

---
_User Notifications_: 
None

### CODE REVIEW

**Comment by idoser96** [comment]
@tomermevorach Tagged for the error messages

**Comment by tomermevorach** [comment]
> @tomermevorach Tagged for the error messages

Thanks! LGTM

File: packages/zendesk-adapter/src/change_validators/default_group_change.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  const groupChanges = changes.filter(isInstanceChange)
+    .filter(change => getChangeData(change).elemID.typeName === GROUP_TYPE_NAME)
+
+  const defaultGroupAddition = groupChanges.filter(isAdditionChange).map(getChangeData)
+    .filter(group => group.value.default === true)
+  const defaultGroupRemoval = groupChanges.filter(isRemovalChange).map(getChangeData)
+    .filter(group => group.value.default === true)
+  const defaultGroupModification = groupChanges.filter(isModificationChange)
+    .filter(change => change.data.before.value.default !== change.data.after.value.default)
+
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/change_validators/default_group_change.ts:59]: nit:
```suggestion
.map(getChangeData)
```

File: packages/zendesk-adapter/src/change_validators/default_group_change.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  return [
+    defaultGroupAddition.map(defaultGroupAdditionError),
+    defaultGroupRemoval.map(defaultGroupRemovalError),
+    defaultGroupModification.map(getChangeData).map(defaultGroupModificationError),
```
Comment (edenhassid) [inline packages/zendesk-adapter/src/change_validators/default_group_change.ts:63]: nit:
```suggestion
    defaultGroupModification.map(defaultGroupModificationError),
```

File: packages/zendesk-adapter/test/change_validators/default_group_change.test.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+const createGroup = (elemName: string, isDefault: boolean): InstanceElement => new InstanceElement(
+  elemName,
+  new ObjectType({ elemID: new ElemID(ZENDESK, GROUP_TYPE_NAME) }),
+  { default: isDefault }
+)
+
+describe('defaultGroupDeletion', () => {
+  const newDefaultGroup = createGroup('newDefaultGroup', true)
+  const removedDefaultGroup = createGroup('removedDefaultGroup', true)
```
Comment (edenhassid) [inline packages/zendesk-adapter/test/change_validators/default_group_change.test.ts:28]: maybe add a remove and addition not default

**Review by edenhassid — COMMENTED** [review]
looks good! added a few nits :)

File: packages/zendesk-adapter/src/change_validators/default_group_change.ts (hunk 1 above)
Comment (idoser96) [reply packages/zendesk-adapter/src/change_validators/default_group_change.ts:59]: Pay attention that i compare the 'before' and the 'after', and getChangeData will only give me the after

Comment (edenhassid) [reply packages/zendesk-adapter/src/change_validators/default_group_change.ts:59]: I meant after the comparison, after the filter

**Review by edenhassid — APPROVED** [review]
🔥 🔥

**Pull request merged**